homeassistant/components/proxmoxve/__init__.py

```diff
@@ -224,7 +224,11 @@ def parse_api_container_vm(status):
     in the future.
     """
 
-    return {"status": status["status"], "name": status["name"]}
+    return {
+        "status": status["status"],
+        "name": status["name"],
+        "mem": status["mem"],
+    }
```

Comment:
I changed the api data parsing to use a dataclass instead. Please let me know if there is something I did wrong with it though, as I haven't used this before.

All of this can most likely be cleaned up a lot more, but I'll save that for after this PR is merged